### DISCUSSION TITLE
How to track the activity lifecycle (onStart, onStop)

### DISCUSSION THREAD

**Discussion by dukerok** [adrielcafe/voyager#170] [Q&A]
Hello. Help me figure out how to track the life cycle of an activity inside Screen. I need to call methods inside a non-compose library, like map.onStart(), map.onStop() using AndoridView

**Comment by Syer10** [comment]
Possibly something like `(LocalContext.current as? LifecycleOwner)`

**Comment by DevSrSouza** [comment]
I'm working on fix the current Voyager LifecycleOwner to support proper Activity `onStart` `onStop` events.

In the mean time, what you can do is something like this:

```kotlin
val LocalActivityLifecycleOwner = staticCompositionLocalOf<LifecycleOwner> {
    error("CompositionLocal LocalActivityLifecycleOwner not present")
}

@Composable
fun YourApplication() {

 CompositionLocalProvider(
    LocalActivityLifecycleOwner provides LocalLifecycleOwner.current
) {
   Navigator(...)
}
```

Then, at your screens, if you want to listen to the Activity Lifecycle events, you just pull `LocalActivityLifecycleOwner.current`

**Reply by pablichjenkov** [reply]
I wonder if the API will have some sort of @Composable
VoyagerApplication {
   staticCompositionLocal<Lifecycle Owner>
  ...
}

Which will take care of that automatically, or we will have to provide the compositionLocal as integration steps, or better both.
Another suggestion is also commonizing it with iOS not just Android.
Looking forward to it.
